### PR TITLE
Tool to extract assessments per registry

### PR DESCRIPTION
**Pull request by marcschier**

### CODE REVIEW

File: tools/clean.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        az group delete -g $group -y
+    fi
+done
+
+# purge deleted keyvault
+#for vault in $(az keyvault list-deleted -o tsv | tr -d '\r'); do
+#    echo "deleting keyvault $vault ..."
+#    az keyvault purge --name $vault
+#done
+
```
Comment (hansgschossmann) [inline tools/clean.sh:65]: i would remove or make it optional

File: tools/clean.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    groups=$(az group list --query "[?starts_with(name, '$prefix')].name" -o tsv | tr -d '\r')
+else
+    # remove groups not marked for keeping
+    groups=$(az group list --query "[?tags.DoNotDelete!='true'].name" -o tsv | tr -d '\r')
+fi
+
+# remove groups 
+for group in $groups; do
+    if [[ $group = MC_* ]] ; then
+        echo "skipping $group ..."
```
Comment (hansgschossmann) [inline tools/clean.sh:51]: interesting

**Review by hansgschossmann — DISMISSED** [review]
:shipit:

File: tools/scripts/acr-scan.ps1 [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    The subscription to use - otherwise uses default.
+
+ .PARAMETER All
+    Include also vulnerabilities that are not patchable.
+#>
+
+Param(
+    [string] $Registry = "industrialiotprod",
+    [string] $Subscription = "IOT_GERMANY",
+    [switch] $All
```
Comment (hansgschossmann) [inline tools/scripts/acr-scan.ps1:25]: not sure if we should expose this information

File: tools/scripts/acr-scan.ps1 [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+$argumentList = @("acr", "show", "--name", $script:Registry, "-ojson")
+$result = (& "az" $argumentList)
+if ($LastExitCode -ne 0) {
+    throw "az $($argumentList) failed with $($LastExitCode)."
+}
+$acr = $result | ConvertFrom-Json
+
+# Get all vulnerability assessments
+$argumentList = @("security", "sub-assessment", "list", "-ojson",
+    "--assessed-resource-id", $acr.id,
```
Comment (hansgschossmann) [inline tools/scripts/acr-scan.ps1:48]: "-o json"? json is default anyhow.

**Review by hansgschossmann — APPROVED** [review]
:shipit:

File: tools/scripts/acr-scan.ps1 (hunk 3 above)
Comment (marcschier) [reply tools/scripts/acr-scan.ps1:25]: It is in several other scripts already.  We should move all these scripts to another repo eventually.

File: tools/scripts/acr-scan.ps1 (hunk 4 above)
Comment (marcschier) [reply tools/scripts/acr-scan.ps1:48]: You can set another default as azure cli configuration, then json is not anymore.

**Pull request merged**